plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/handlers/SQLEditorHandlerExecute.java

```diff
@@ -52,6 +59,32 @@ public Object execute(ExecutionEvent event) throws ExecutionException
             case SQLEditorCommands.CMD_EXECUTE_SCRIPT:
                 editor.processSQL(false, true);
                 break;
+            case SQLEditorCommands.CMD_EXECUTE_SCRIPT_NATIVE:
+                SQLNativeExecutorDescriptor executorDescriptor = SQLNativeExecutorRegistry.getInstance()
+                    .getExecutorDescriptor(editor.getDataSource());
+                if (executorDescriptor == null) {
+                    throw new ExecutionException("Valid native executor is not found");
+                }
+                try {
+                    if (editor.getExecutionContext() instanceof DBCExecutionContextDefaults) {
+                        DBCExecutionContextDefaults<?, ?> executionContext
+                            = (DBCExecutionContextDefaults<?, ?>) editor.getExecutionContext();
+                        SQLEditorExecutor<DBSObject> nativeExecutor
+                            = (SQLEditorExecutor<DBSObject>) executorDescriptor.getNativeExecutor();
+                        if (nativeExecutor == null) {
+                            throw new ExecutionException("Valid native is executor not found");
+                        }
+                        DBSObject object = executionContext.getDefaultCatalog();
+                        if (object == null) {
+                            object = editor.getDataSource();
+                        }
+                        nativeExecutor.execute(object, editor);
+                    }
+
+                } catch (DBException e) {
+                    log.error(e);
+                }
+                break;
```

Comment:
You're using local variables inside a switch case. Please wrap it with braces to avoid polluting the scope:

```suggestion
            case SQLEditorCommands.CMD_EXECUTE_SCRIPT_NATIVE: {
                SQLNativeExecutorDescriptor executorDescriptor = SQLNativeExecutorRegistry.getInstance()
                    .getExecutorDescriptor(editor.getDataSource());
                if (executorDescriptor == null) {
                    throw new ExecutionException("Valid native executor is not found");
                }
                try {
                    if (editor.getExecutionContext() instanceof DBCExecutionContextDefaults) {
                        DBCExecutionContextDefaults<?, ?> executionContext
                            = (DBCExecutionContextDefaults<?, ?>) editor.getExecutionContext();
                        SQLEditorExecutor<DBSObject> nativeExecutor
                            = (SQLEditorExecutor<DBSObject>) executorDescriptor.getNativeExecutor();
                        if (nativeExecutor == null) {
                            throw new ExecutionException("Valid native is executor not found");
                        }
                        DBSObject object = executionContext.getDefaultCatalog();
                        if (object == null) {
                            object = editor.getDataSource();
                        }
                        nativeExecutor.execute(object, editor);
                    }
                } catch (DBException e) {
                    log.error(e);
                }
                break;
            }
```